### ISSUE TITLE
Custom version of spark

### ISSUE THREAD

**Issue by babak12** [andypetrella/spark-notebook#364]
Hi, I am going to build the spark-notebook for a custom version of spark which it is source on a local drive. What procedures should I take to build the notebook?

**Comment by RPCMoritz** [comment]
From what I understand spark is a run-time dependency, so you can build against an API-version, and then supply the path to your spark-lib in the configuration.
Otherwise you can reference a specific spark version at build-time and deploy your spark-build to your local .ivy2. Then it should be pulled during the build.

**Comment by andypetrella** [comment]
indeed, although this will change in further versions. the easiest right now is to use a clone and sbt. 

you publish your spark locally, let's say version 1.6.0-SNAPSHOT.

then you can use it via sbt like this 
```
sbt -Dspark.version=1.6.0-SNAPHSOT
```

**however** there is working crapy to do, and I'm sad I didn't spent the time yet for a cleaner solution... you'll have to duplicate the source directory in the spark module for your spark version (without SNAPSHOT,  though) and the scala version... I'm sorry about that, really
